packages/express-middleware/README.md

```diff
@@ -0,0 +1,129 @@
+# @loopback/express-middleware
+
+[Express middleware](https://expressjs.com/en/guide/using-middleware.html) are
+key building blocks for Express applications. Simple APIs are provided by
+Express to set up middleware. It works for applications where the middleware
+chain registration can be centralized, such as:
+
+```js
+app.use(middlewareHandler1);
+app.use(middlewareHandler2);
+// ...
+app.use(middlewareHandlerN);
+```
+
+The middleware handlers will be executed by Express in the order of `app.use()`
+is called. There are obvious limitations of this approach if your application
+allows multiple modules to contribute middleware.
+
+1. It's hard to align middleware in the right order
+2. It's hard to separate configuration of middleware
+
+This module introduces a simple extension that utilizes LoopBack 4's inversion
+of controller (Ioc) dependency injection (DI) container offered by
+`@loopback/context`.
+
+## Overview
+
+We introduce `ExpressContext` to manage middleware registration and provide a
+handler function for Express applications.
+
+1. `ExpressContext` allows us to bind middleware handlers with optional settings
+   such as `path` and `phase` as tags. Each binding is tagged with `middleware`
+   so that it can be discovered.
+
+2. `ExpressContext` sorts the middleware entries using `phase`. The order of
+   phases can be configured.
+
+3. `ExpressContext` exposes `requestHandler` function which is an Express
+   middleware handler function that can be mounted to Express applications.
+
+4. `ExpressContext` observes binding events to keep track of latest list of
+   middleware and rebuilds the chain if necessary.
+
+## Installation
+
+To use this package, you'll need to install `@loopback/express-middleware`.
+
+```sh
+npm i @loopback/express-middleware
+```
+
+## Basic Use
+
+### Register middleware handler
+
+Handler functions compatible with Express middleware can be registered as
+follows:
+
+```ts
+const expressCtx = new ExpressContext();
+expressCtx.middleware(cors(), {phase: 'cors'});
```

Comment:
Do you expect users to call `middleware` with a single argument only? If not, then I think the following API may be more ergonomic:

```ts
expressCtx.middleware({
  handler: cors(),
  phase: 'cors',
  // path, etc.
});
```